src/mem/ruby/network/MessageBuffer.cc

```diff
@@ -205,7 +205,7 @@ MessageBuffer::peek() const
 Tick
 random_time()
 {
-    static Random::RandomPtr rng = Random::genRandom();
+    Random::RandomPtr rng = Random::genRandom();
```

Comment:
is this not going to generate a new heap instance of `Random::Random` each time `random_time` is called? We should create one instance and reuse it (e.g., with `static`). At least that is my understanding of the original code.